text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,623 @@
+  details. This team must have at least 2 developers.
+  - The target maintainers should not only fix target-specific issues, but
+    should use any such issue as an opportunity to educate the Rust community
+    about portability to their target, and enhance their documentation of the
+    target.
+- The target must not place undue burden on Rust developers not specifically
+  concerned with that target. Rust developers are expected to not gratuitously
+  break a tier 2 target, but are not expected to become experts in every tier 2
+  target, and are not expected to provide target-specific implementations for
+  every tier 2 target.
+- The target must provide documentation for the Rust community explaining how
+  to build for the target using cross-compilation, and explaining how to run
+  tests for the target. If at all possible, this documentation should show how
+  to run Rust programs and tests for the target using emulation, to allow
+  anyone to do so. If the target cannot be feasibly emulated, the documentation
+  should explain how to obtain and work with physical hardware, cloud systems,
+  or equivalent.
+- The target must document its baseline expectations for the features or
+  versions of CPUs, operating systems, libraries, runtime environments, and
+  similar.
+- If introducing a new tier 2 or higher target that is identical to an existing
+  Rust target except for the baseline expectations for the features or versions
+  of CPUs, operating systems, libraries, runtime environments, and similar,
+  then the proposed target must document to the satisfaction of the approving
+  teams why the specific difference in baseline expectations provides
+  sufficient value to justify a separate target.
+  - Note that in some cases, based on the usage of existing targets within the
+    Rust community, Rust developers or a target's maintainers may wish to
+    modify the baseline expectations of a target, or split an existing target
+    into multiple targets with different baseline expectations. A proposal to
+    do so will be treated similarly to the analogous promotion, demotion, or
+    removal of a target, according to this policy, with the same team approvals
+    required.
+    - For instance, if an OS version has become obsolete and unsupported, a
+      target for that OS may raise its baseline expectations for OS version
+      (treated as though removing a target corresponding to the older
+      versions), or a target for that OS may split out support for older OS
+      versions into a lower-tier target (treated as though demoting a target
+      corresponding to the older versions, and requiring justification for a
+      new target at a lower tier for the older OS versions).
+- Tier 2 targets must not leave any significant portions of `core` or the
+  standard library unimplemented or stubbed out, unless they cannot possibly be
+  supported on the target.
```

Comment:
This doesn't need to be addressed here, but for new targets like WebAssembly which are not only new to the Rust project but new in general, there are significant portions of the std lib that cannot _currently_ be implemented because the target itself doesn't yet have the functionality but this will change over time.